hibernate-reactive-core/src/main/java/org/hibernate/reactive/engine/impl/EntityTypes.java

```diff
@@ -124,13 +134,29 @@ static CompletionStage<Object> loadByUniqueKey(
 		else {
 			return persister
 					.reactiveLoadByUniqueKey( uniqueKeyPropertyName, key, session )
-					.thenApply( loaded -> {
+					.thenApply( ukResult -> {
+						Object loadedUniqueKey = ukResult;
+						// The unique key may be a HibernateProxy object and needs to be loaded before
+						// adding to persistenceContext
+						if ( loadedUniqueKey instanceof HibernateProxy ) {
+							try {
+								CompletionStage<Object> resultObject = loadHibernateProxyEntity(
+										(HibernateProxy) loadedUniqueKey,
+										session
+								);
+								loadedUniqueKey = ((CompletableFuture)resultObject).get();
+							}
```

Comment:
You should not use a `.get()` because it's a blocking method.
 Assuming the logic is correct (I haven't cheked yet), the code should look something like:
```
.thenCompose ukResult -> {
      ...
      if ( loadedUniqueKey instanceof HibernateProxy ) {
         return   loadHibernateProxyEntity( ... )
                            .thenApply(loadedUniqueKey -> { .... } );
      }  
      ... 
}
```